src/db/connection.rs

```diff
@@ -0,0 +1,106 @@
+use crate::db::sql_types::*;
+use diesel::pg::PgConnection;
+use rocket::fairing::Fairing;
+use rocket_contrib::databases::r2d2::{Pool, PooledConnection};
+use rocket_contrib::databases::Poolable;
+
+type Connection = PooledConnection<<PgConnection as Poolable>::Manager>;
+
+/// The request guard type.
+pub struct RequestContext {
+    pub conn: Connection,
+}
+
+/// The pool type.
+pub struct ConnectionPool(Pool<<PgConnection as Poolable>::Manager>);
+
+impl From<Connection> for RequestContext {
+    fn from(conn: Connection) -> Self {
+        Self { conn }
+    }
+}
+impl RequestContext {
+    /// Get a DatabaseContext from this struct that can be used to connect
+    /// to the db
+    pub fn database_context(
+        &self,
+        action: ActionType,
+        modifier: ActionModifier,
+    ) -> DatabaseContext {
+        return DatabaseContext::from(&self.conn, action, modifier);
+    }
+
+    /// Returns a fairing that initializes the associated database
+    /// connection pool.
+    pub fn fairing() -> impl Fairing {
+        ::rocket::fairing::AdHoc::on_attach("\'postgres\' Database Pool", |rocket| {
```

Comment:
Pretty sure the leading `::` isn't necessary

Reply:
fixed!